hack/generator/pkg/config/export_filter.go

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package config
+
+// ExportFilterAction defines the possible actions that should happen for types matching the filter
+type ExportFilterAction string
+
+const (
+	// ExportFilterActionInclude indicates that any type matched by the filter should be exported to disk by the generator
+	ExportFilterActionInclude ExportFilterAction = "include"
```

Comment:
The prefix here was because I was struggling with conflicting enums and this was _a way_ to differentiate different enums in the same package. If anybody has a better way I'm open to it

Reply:
I'm not sure how golangy it is but you could make a very small package so that usage is like `exportfilter.Include`? @babbageclunk is that a done thing?

Reply:
Yeah I'm not really sure what to do here -- totally open to ideas.

Reply:
I'm conflicted. I agree they should be separate enums (because they have different reasons to change), but the current naming is clunky. Moving to a different package wouldn't really help, unless you created several - one per enum, and that's even worse.

Delphi had a similar problem - unlike C# where enum values are namespaced within the type, Delphi enum values were global identifiers. The common idiom was to use a two letter, lowercase, prefix as a pseudo namespace.

Since filter actions are within-package, we could do the same ... so for ExportFilterAction, use `ea`:

``` go
eaInclude ExportFilterAction = "include"
eaExclude ExportFilterAction = "exclude"
```

and for `TypeFilterAction`, use `ta`:

``` go
taInclude TypeFilterAction = "include"
taPrune TypeFilterAction = "prune"
```

Reply:
horizontal space isn't really at a premium though, so why bother with the abbreviations at all?

`eaInclude` is significantly less clear (at the usage site) than `exportFilterActionInclude` (or `exportFilterInclude` if we prefer that), or `exportfilter.include` if we do the package-per-enum thing George is mentioning (which does suck because I think you can only have 1 package per directory so we'd have to pull it off into its own directory which I don't like).

Reply:
I think `exportFilterInclude` and `exportFilterExclude` are probably the best option here - `exportFilterActionInclude` feels like it crosses some psychological word-complexity boundary. It doesn't really merit a small package, given the whole having to put it in a separate directory thing. You could use an anonymous type like this:

```go
type ExportFilterAction string

var ExportFilter = struct {
	Include ExportFilterAction
	Exclude ExportFilterAction
}{
	Include: "include",
	Exclude: "exclude",
}
```

... and that's not *too* bad - it's ok to use at least. It does require having something named `ExportFilter` that isn't an export filter though. And they're not constants either, which is dicey.

Reply:
Oh, and then you've got `ExportFilter` just below. Anyway, I was just including that as a "you could do this but don't" :).

Reply:
I agree about the psychological word-complexity boundary for sure. I'll update the pattern for these two to be `exportFilterInclude`/`exportFilterExclude` and the equivalent for `typeFilter`.